Octokit.Tests.Integration/Clients/Enterprise/EnterpriseManagementConsoleClientTests.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Threading.Tasks;
+using Octokit;
+using Octokit.Tests.Integration;
+using Xunit;
+
+public class EnterpriseManagementConsoleClientTests
+{
+    readonly IGitHubClient _github;
+
+    public EnterpriseManagementConsoleClientTests()
+    {
+        _github = EnterpriseHelper.GetAuthenticatedClient();
+    }
+
+    [GitHubEnterpriseTest]
+    public async Task CanGetMaintenanceMode()
+    {
+        var maintenance = await _github.Enterprise.ManagementConsole.GetMaintenanceMode("Password01");
+
+        Assert.NotNull(maintenance);
+    }
+
+    [GitHubEnterpriseTest]
+    public async Task CanSetMaintenanceModeOff()
+    {
+        // Set maintenance mode OFF now
+        var maintenance = await
+            _github.Enterprise.ManagementConsole.EditMaintenanceMode(
+                new UpdateMaintenanceRequest(),
+                EnterpriseHelper.ManagementConsolePassword);
+
+        Assert.NotNull(maintenance);
+        Assert.Equal(maintenance.Status, MaintenanceModeStatus.Off);
+    }
+
+    [GitHubEnterpriseTest]
+    public async Task CanSetMaintenanceModeOnNow()
+    {
+        // Set maintenance mode ON now
+        var maintenance = await
+            _github.Enterprise.ManagementConsole.EditMaintenanceMode(
+                new UpdateMaintenanceRequest(true, MaintenanceDate.Now()),
+                EnterpriseHelper.ManagementConsolePassword);
+
+        Assert.NotNull(maintenance);
+        Assert.Equal(maintenance.Status, MaintenanceModeStatus.On);
+
+        // Ensure maintenance mode is OFF
+        await _github.Enterprise.ManagementConsole.EditMaintenanceMode(
```

Comment:
We don't seem to be doing any asserting here? Will the request will fail here if the GHE server is not in the right state?